forte/mrdsrg-spin-integrated/dsrg_mrpt2.h

```diff
@@ -175,6 +178,190 @@ class DSRG_MRPT2 : public MASTER_DSRG {
     /// Unitary matrix to block diagonal Fock
     ambit::BlockedTensor U_;
 
+
+    // NOTICE These are essential variables and functions for computing DSRG-MRPT2 gradient.
+    // Some variables may be redundant thus need further elimination
+    /// Set ambit tensor labels
+    void set_ambit_space();
+    /// Set tensors
+    void set_tensor();
+    /// Set density
+    void set_density();
+    /// Set Fock matrix
+    void set_fock();
+    /// Set Hamiltonian
+    void set_h();
+    /// Set two-electron integrals
+    void set_v();
+    /// Set CI-relevant integrals
+    void set_ci_ints();
+
+    /// Set MO space environment and global variables
+    void set_global_variables();
+    size_t nmo_;
+    size_t ncore_;
+    size_t nvirt_;
+    size_t na_;
+    size_t nirrep_;
+
+
+    void set_dsrg_tensor();
+
+   
+    /// Write the Lagrangian
+    void write_lagrangian();
+    /// Write spin_dependent one-RDMs coefficients
+    void write_1rdm_spin_dependent();
+    /// Write spin_dependent two-RDMs coefficients using IWL
+    void write_2rdm_spin_dependent();
+    /// TPDM backtransform
+    void tpdm_backtransform();
+
+   
+    void set_multiplier(); 
+
+    void solve_z();
+    void set_sigma();
+    void set_xi();
+    void set_tau();
+    void set_kappa();
+    void set_z();
+    void set_alpha();
+    void set_CI();
+    void set_b();
+    void set_z_cc();
+    void set_z_vv();
+    void set_z_aa_diag();
+    void set_w();
+
+    double s;
+
+    size_t ndets;
+
+    /// List of core MOs (Correlated)
+    std::vector<size_t> core_mos_;
+    /// List of active MOs (Correlated)
+    std::vector<size_t> actv_mos_;
+    /// List of virtual MOs (Correlated)
+    std::vector<size_t> virt_mos_;
+    /// List of core MOs (Absolute)
+   
+    std::vector<size_t> core_all_;
+    /// List of active MOs (Absolute)
+    std::vector<size_t> actv_all_;
+    std::vector<size_t> virt_all_;
+
+    /// List of relative core MOs
+    std::vector<std::pair<unsigned long, unsigned long>,
+                std::allocator<std::pair<unsigned long, unsigned long>>>
+        core_mos_relative;
```

Comment:
Please follow the naming convention where class variables are ended with `_`.

Reply:
fixed